### DISCUSSION TITLE
`CpScreenResponseBehavior::content` returns an empty string

### DISCUSSION THREAD

**Discussion by rynpsc** [craftcms/cms#13002] [General]
When trying to get the current value of `$response->content` from `prepareEditScreen` an empty string is returned.

```php
public function prepareEditScreen(Response $response, string $containerId): void
{
    Craft::dd($response->content); // null
}
```

However you can get the current value if you go via the behaviour itself.

```php
public function prepareEditScreen(Response $response, string $containerId): void
{
    Craft::dd($response->getBehavior('cp-screen')->content); // <input type="hidden" name="elementId" value="521"> ...
}
```

Is this expected behaviour? Is there a better/correct way of accessing the value to modify it?

**Comment by brandonkelly** [comment]
It’s not working is because `yii\web\Response` defines its own `content` property, which is used by response formatters to define the final response body. So `$response->content` is resolving to that, rather than falling back to the behavior’s `content` property.

I hadn’t considered that the content may need to be modified after being set. We could possibly rename `CpScreenResponseBehavior::content` to `contentHtml` (etc.) in Craft 5 to avoid the name conflict. We’ll discuss that internally.

**Reply by brandonkelly** [reply]
That `content` → `contentHtml` change has been made for Craft 5, along with all other HTML properties/methods on `CpScreenResponseBehavior` (#13037).
